docs/sapling-comparison.md

```diff
@@ -0,0 +1,70 @@
+  user explicitly tells the tool when to create a commit and which files to
+  include. When using jj, the working copy
+  is [automatically snapshotted by every command](working-copy.md). New files
+  are automatically tracked and deleted files are automatically untracked. This
+  has several advantages:
+    * The working copy is effectively backed up every time you run a command.
+    * No commands fail because you have changes in the working copy ("abort: 1
+      conflicting file changes: ..."). No need for `sl shelve`.
+    * Simpler and more consistent CLI because the working copy is treated like
+      any other commit.
+* **Conflicts:** Like most VCSs, Sapling requires the user to
+  resolve conflicts before committing. jj lets
+  you [commit conflicts](conflicts.md). Note that it's a representation of the
+  conflict that's committed, not conflict markers (`<<<<<<<` etc.). This also
+  has several advantages:
+    * Merge conflicts won't prevent you from checking out another commit.
+    * You can resolve the conflicts when you feel like it.
+    * Rebasing descendants always succeeds. Like jj, Sapling automatically
+      rebases, but it will fail if there are conflicts.
+    * Merge commits can be rebased correctly (Sapling sometimes fails).
+    * You can rebase conflicts and conflict resolutions.
+* **Undo:** jj's undo is powered by [the operation log](operation-log.md), which
+  records how the repo has changed over time. Sapling has a similar feature
+  with its [MetaLog](https://sapling-scm.com/docs/internals/metalog).
+  They seem to provide similar functionality, but jj also exposes the log to the
+  user via `jj op log`, so you can tell how far back you want to go back.
+  Sapling has `sl debugmetalog`, but that seems to show the history of a single
+  commit, not the whole repo's history. Thanks to jj snapshotting the working
+  copy, it's possible to undo changes to the working copy. For example, if
+  you `jj undo` a ` jj commit`, `jj diff` will show the same changes as
+  before `jj commit`, but if you `sl undo` a `sl commit`, the working copy will
+  be clean.
+* **Git interop:** Sapling supports cloning, pushing, and pulling from a remote
+  Git repo. jj also does, and it also supports sharing a working copy with a Git
+  repo, so you can use `jj` and `git` interchangeably in the same repo.
+* **Polish:** Sapling is much more polished and feature-complete. For example,
+  jj has no `blame/annotate` or `bisect` commands, and also no copy/rename
+  support. Sapling also has very nice web UI
+  called [Interactive Smartlog](https://sapling-scm.com/docs/addons/isl), which
+  lets you drag and drop commits to rebase them, among other things.
+* **Forge integrations:** Sapling has `sl pr submit --stack`, which lets you
+  push a stack of commits as separate GitHub PRs (or GitLab MRs, since it simply
+  creates branches?). You will then have to manually add the base branch for
```

Comment:
> You will then have to manually add the base branch for each PR in GitHub's UI

Sorry, I don't follow one. Do you mean users need to do this when merging the PRs or creating PRs?

Reply:
> Sorry, I don't follow this line. Do you mean users need to do this when merging the PRs or creating PRs?

When creating PRs.

> Not sure if you are aware of the various workflows supported by `sl pr submit`.
> 
> https://github.com/facebook/sapling/blob/a66a40a339e939aa2000d849137799edd0cbddc4/eden/scm/edenscm/ext/github/submit.py#L39-L65

Thanks, I wasn't at all aware (I have never tried `sl pr submit`, so what I wrote here was just based on what I found on https://sapling-scm.com/docs/git/sapling-stack. That's nice that `sl pr submit` updates the base branches! I've updated the text here.

Reply:
Yeah, we should update the document of `sl pr submit`.